src/test/java/hudson/matrix/MatrixProjectTest.java

```diff
@@ -412,6 +413,24 @@ public void testTwoFileParams() throws Exception {
         j.assertBuildStatusSuccess(f.get(10,TimeUnit.SECONDS));
     }
 
+    @Issue("JENKINS-34758")
+    @Test
+    public void testParametersAsEnvOnChildren() throws Exception {
+        MatrixProject p = createMatrixProject();
+        p.setAxes(new AxisList(new TextAxis("foo","1")));
+        p.addProperty(new ParametersDefinitionProperty(
+            new StringParameterDefinition("MY_PARAM","")
+        ));
+        // must fail if $MY_PARAM or $foo are not defined in children
+        p.getBuildersList().add(new Shell("set -eux; echo $MY_PARAM; echo $foo"));
```

Comment:
Why would be that better? Currently if either `$MY_PARAM` or `$foo` are not defined the build fails, so if the build finishes successfully it is proving that both variables have been injected properly (which is the aim of this test).